tests/Aspire.Dashboard.Tests/Integration/DashboardClientAuthTests.cs

```diff
@@ -0,0 +1,177 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using Aspire.Dashboard.Configuration;
+using Aspire.Dashboard.Model;
+using Google.Protobuf.WellKnownTypes;
+using Aspire.ResourceService.Proto.V1;
+using Grpc.Core;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Xunit;
+using DashboardServiceBase = Aspire.ResourceService.Proto.V1.DashboardService.DashboardServiceBase;
+
+namespace Aspire.Dashboard.Tests.Integration;
```

Comment:
These tests are ok and don't need to be rewritten, but I think there is a simpler implementation. You could have added support for passing a custom `HttpMessageHandler` to the resource client, then capture the outgoing request when it is sent to the handler and returned a response from the handler. Then assert the outgoing request has headers expected. That saves having a mock server. The integration tests could turn into unit tests.

On the other hand, binary gRPC request/responses aren't easily approachable when working at a bytes level like a message handler does.